File: packages/web/src/api/materialCheck.ts
import api from "./index";

export interface MaterialCheckItem {
  id: number;
  workorder_id: number;
  bom_item: string;
  detected_material: string;
  result: string;
  confidence: number;
  snapshot_url: string;
  detail: string;
  checked_at: string;
}

export interface MaterialCheckStats {
  total: number;
  ok_count: number;
  ng_count: number;
  warn_count: number;
  pass_rate: number;
}

export interface MaterialCheckListResult {
  items: MaterialCheckItem[];
  total: number;
}

export const materialCheckApi = {
  list: (params?: {
    workorder_id?: number;
    result?: string;
    bom_item?: string;
    skip?: number;
    limit?: number;
  }) => api.get<MaterialCheckListResult>("/material-check/", { params }),
  create: (data: Omit<MaterialCheckItem, "id" | "checked_at">) =>
    api.post<MaterialCheckItem>("/material-check/", data),
  get: (id: number) => api.get<MaterialCheckItem>(`/material-check/${id}`),
  byWorkorder: (workorderId: number) =>
    api.get<MaterialCheckItem[]>(`/material-check/workorder/${workorderId}`),
  stats: (params?: { workorder_id?: number }) =>
    api.get<MaterialCheckStats>("/material-check/stats", { params }),
};

File: packages/server/src/schemas/material_check.py
from datetime import datetime

from pydantic import BaseModel, Field


class MaterialCheckCreate(BaseModel):
    workorder_id: int
    bom_item: str = Field(..., max_length=100)
    detected_material: str = ""
    result: str = Field(..., pattern="^(OK|NG|WARN)$")
    confidence: float = Field(default=0.0, ge=0, le=1)
    snapshot_url: str = ""
    detail: str = ""


class MaterialCheckResponse(BaseModel):
    id: int
    workorder_id: int
    bom_item: str
    detected_material: str
    result: str
    confidence: float
    snapshot_url: str
    detail: str
    checked_at: datetime

    model_config = {"from_attributes": True}


class MaterialCheckStats(BaseModel):
    total: int = 0
    ok_count: int = 0
    ng_count: int = 0
    warn_count: int = 0
    pass_rate: float = 0.0


class MaterialCheckListResponse(BaseModel):
    items: list[MaterialCheckResponse]
    total: int
